src/plugins/intel_gpu/include/intel_gpu/graph/serialization/binary_buffer.hpp

```diff
@@ -19,9 +19,8 @@ class BinaryOutputBuffer : public OutputBuffer<BinaryOutputBuffer> {
 
     void write(void const * data, std::streamsize size) {
         auto const written_size = stream.rdbuf()->sputn(reinterpret_cast<const char*>(data), size);
-        if (written_size != size) {
-            throw std::runtime_error("Failed to write " + std::to_string(size) + " bytes to stream! Wrote " + std::to_string(written_size));
-        }
+        OPENVINO_ASSERT(written_size == size,
+            "Failed to write " + std::to_string(size) + " bytes to stream! Wrote " + std::to_string(written_size));
```

Comment:
https://github.com/openvinotoolkit/openvino/pull/13801#discussion_r1017891613
Changed to use `OPENVINO_ASSERT`. Thank you.